object_intern.go

```diff
@@ -247,28 +347,29 @@ func (oi *ObjectIntern) AddOrGetString(obj []byte, safe bool) (string, error) {
 //
 // This method does not increase the reference count of the interned object.
 func (oi *ObjectIntern) GetPtrFromByte(obj []byte) (uintptr, error) {
-	var addr gos.ObjAddr
-	var ok bool
-	var objString string
-
 	if oi.conf.Compression != None {
-		objString = string(oi.compress(obj))
-	} else {
-		objString = string(obj)
+		oi.RLock()
```

Comment:
Instead of many `oi.RLock()` / `oi.RUnlock()` calls, this could simply have each once at the beginning like
```
oi.RLock()
defer oi.RUnlock()
```

Reply:
Yes, I was trying to see which would be faster at some point. I still need to run it through benchmarks (after I boot into a suitable environment for that)

Reply:
https://github.com/robert-milan/go-object-interning/issues/15#issuecomment-511092574

Too large of a performance hit